source/includes/usage-examples/retrieve-code-examples.c

```diff
@@ -0,0 +1,113 @@
+#include <bson/bson.h>
+#include <mongoc/mongoc.h>
+#include <stdio.h>
+
+int
+main (int argc, char *argv[])
+{
+    mongoc_client_t *client;
+    mongoc_collection_t *collection;
+    mongoc_cursor_t *results;
+    char *str;
+    bson_error_t error;
+
+    mongoc_init ();
+
+    client =
+        mongoc_client_new ("<connection string URI>");
+    collection = mongoc_client_get_collection (client, "<database name>", "collection name");
+
+    // start-find
+    {
+        bson_t *query = bson_new ();
+        // Add fields to query here
+        results = mongoc_collection_find_with_opts (collection, query, NULL, NULL);
+        const bson_t *doc;
+
+        while (mongoc_cursor_next (results, &doc)) {
+            str = bson_as_canonical_extended_json (doc, NULL);
+            printf ("%s\n", str);
+            bson_free (str);
+        }
+
+        mongoc_cursor_destroy (results);
+        bson_destroy (query);
+    }
+    // end-find
+
+    // start-count-all
+    {
+        int64_t count = 
+            mongoc_collection_count_documents (collection, bson_new (), NULL, NULL, NULL, &error);
+
+        printf ("%" PRId64 "\n", count);
```

Comment:
```suggestion
        bson_t *query = bson_new();
        int64_t count =
            mongoc_collection_count_documents(collection, query, NULL, NULL, NULL, &error);

        printf("%" PRId64 "\n", count);
        bson_destroy(query);
```

To avoid a memory leak. `bson_new()` returns an allocated `bson_t`.

Reply:
I can make this change. Question: any reason that we prefer defining query on its own line?